### ISSUE TITLE
why random action?

### ISSUE THREAD

**Issue by sezan92** [MorvanZhou/Reinforcement-learning-with-tensorflow#73]
I was going through your code. I noticed this one
![random action](https://user-images.githubusercontent.com/11025093/43618534-df4a1da2-9703-11e8-9ac3-fb0cb9589359.png)

Why are you using np.random.choice? shouldnt we take action of maximum probability ?

**Comment by MorvanZhou** [comment]
Hi @sezan92 ， please tell me which code file you refer to?

**Comment by sezan92** [comment]
https://github.com/MorvanZhou/Reinforcement-learning-with-tensorflow/blob/a3ee123b57a12008c178962414e02947484d741d/contents/8_Actor_Critic_Advantage/AC_CartPole.py#L81

**Comment by MorvanZhou** [comment]
In Actor-Critic, we the actor loss is based on action probability rather than the action itself in Q-learning. Therefore, I choose to output probability. The definition of loss in the Actor can be found at here: https://github.com/MorvanZhou/Reinforcement-learning-with-tensorflow/blob/a3ee123b57a12008c178962414e02947484d741d/contents/8_Actor_Critic_Advantage/AC_CartPole.py#L66-L67

**Comment by sezan92** [comment]
I got it you chose the probability as output. but we need to choose the action with maximum probability right? but you choose a random sample from probabilities of action , not the maximum probability. my question was, why choosing random sample ?

**Comment by MorvanZhou** [comment]
In Actor-Critic, we use probability rather than maximum action value. In Q learning, we use maximum action. These two are different.

**Comment by sezan92** [comment]
no, i am not talking about taking the action for maximum Q value. the last layer of the actor network is softmax. hence it will give probability of every action. so the action with maximum probability - which is different than action with maximum Q value- should be chosen , right? thats what I am talking about

**Comment by cyberphantom** [comment]
I have the same concern as you. I understand the code ad the math behind it but only this part. I read the comments and I still didn't get it. How is the right action is going to be selected if we keep selecting random choice. The code is correct and the network is converging but still I'm confused. IF anybody have an answer please help
